### ISSUE TITLE
[Bug] (function) The behavior of str_to_date function is inconsistent between Be and Fe

### ISSUE THREAD

**Issue by xy720** [apache/doris#15777]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```
mysql> select str_to_date('202301', '%Y%m');
+-------------------------------+
| str_to_date('202301', '%Y%m') |
+-------------------------------+
| 2023-01-00                    |
+-------------------------------+
1 row in set (0.00 sec)

mysql> select str_to_date('202301', '%Y%m') from test_2;
+-------------------------------+
| str_to_date('202301', '%Y%m') |
+-------------------------------+
| 0000-00-00                    |
| 0000-00-00                    |
+-------------------------------+
2 rows in set (0.01 sec)
```

### What You Expected?

Let them be consistent.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by xy720** [comment]
@Gabriel39 Please have a look at this?

**Comment by Gabriel39** [comment]
Sure. I will fix it in 1.2.2/2.0.0
